### PR TITLE
Simplify handling of secret key

### PR DESCRIPTION
**Pull request by nscuro**
As of Dependency-Track 4.7.0 (which I just rebased our [`kafka-poc`](https://github.com/sahibamittal/dependency-track/tree/kafka-poc) branch against), the secret key can now be provided in a generic byte-encoded format, and the location of it is configurable, see https://docs.dependencytrack.org/getting-started/configuration/#secret-key

This removes the necessity for various workarounds we had in place to allow for sharing of the secret key amongst API server and services like the repo meta analyzer.

### CODE REVIEW

File: commons/src/main/java/org/acme/common/SecretDecryptor.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+        // Generating IV
+        int ivSize = 16;
+        final byte[] iv = new byte[ivSize];
+        final SecureRandom random = new SecureRandom();
+        random.nextBytes(iv);
+        final IvParameterSpec ivParameterSpec = new IvParameterSpec(iv);
+
+        // Encrypt
+        final Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
```
[the remark this answers is not in the document]
Comment (nscuro) [reply commons/src/main/java/org/acme/common/SecretDecryptor.java:82]: Can't do anything about this as it must be compatible with what DT is doing currently.

[the remark this answers is not in the document]
Comment (nscuro) [reply commons/src/main/java/org/acme/common/SecretDecryptor.java:82]: @sonatype-lift ignoreall

File: commons/src/main/java/org/acme/common/SecretDecryptorConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    @Produces
+    @ApplicationScoped
+    SecretDecryptor secretDecryptor(final SecretKey secretKey) {
+        return new SecretDecryptor(secretKey);
+    }
+
+    @Produces
+    @ApplicationScoped
+    SecretKey secretKey(@ConfigProperty(name = "secret.key.path", defaultValue = "${HOME}/.dependency-track/keys/secret.key") final String secretKeyPath) throws IOException {
+        try (final InputStream fis = Files.newInputStream(Paths.get(secretKeyPath))) {
```
[the remark this answers is not in the document]
Comment (nscuro) [reply commons/src/main/java/org/acme/common/SecretDecryptorConfiguration.java:25]: It is indeed intended that this path is user-configurable.

**Review by mehab — APPROVED** [review]
lgtm

**Pull request merged**